whc-shared/src/main/java/com/ibm/whc/deid/shared/pojo/config/masking/BinningMaskingProviderConfig.java

```diff
@@ -17,99 +17,169 @@
+		this.useStartValue = useStartValue;
+	}
+
+	public boolean isUseSingleBucketOverThreshold() {
+		return useSingleBucketOverThreshold;
+	}
+
+	public void setUseSingleBucketOverThreshold(boolean useSingleBucketOverThreshold) {
+		this.useSingleBucketOverThreshold = useSingleBucketOverThreshold;
+	}
+
+	public double getSingleBucketOverThresholdValue() {
+		return singleBucketOverThresholdValue;
+	}
+
+	public void setSingleBucketOverThresholdValue(double singleBucketOverThresholdValue) {
+		this.singleBucketOverThresholdValue = singleBucketOverThresholdValue;
+	}
+
+	public String getSingleBucketOverThresholdReplacement() {
+		return singleBucketOverThresholdReplacement;
+	}
+
+	public void setSingleBucketOverThresholdReplacement(String singleBucketOverThresholdReplacement) {
+		this.singleBucketOverThresholdReplacement = singleBucketOverThresholdReplacement;
+	}
+
+	public boolean isUseSingleBucketUnderThreshold() {
+		return useSingleBucketUnderThreshold;
+	}
+
+	public void setUseSingleBucketUnderThreshold(boolean useSingleBucketUnderThreshold) {
+		this.useSingleBucketUnderThreshold = useSingleBucketUnderThreshold;
+	}
+
+	public double getSingleBucketUnderThresholdValue() {
+		return singleBucketUnderThresholdValue;
+	}
+
+	public void setSingleBucketUnderThresholdValue(double singleBucketUnderThresholdValue) {
+		this.singleBucketUnderThresholdValue = singleBucketUnderThresholdValue;
+	}
+
+	public String getSingleBucketUnderThresholdReplacement() {
+		return singleBucketUnderThresholdReplacement;
+	}
+
+	public void setSingleBucketUnderThresholdReplacement(String singleBucketUnderThresholdReplacement) {
+		this.singleBucketUnderThresholdReplacement = singleBucketUnderThresholdReplacement;
+	}
+	
+	@Override
+	public void validate(DeidMaskingConfig maskingConfig) throws InvalidMaskingConfigurationException {
+		super.validate(maskingConfig);
+		if (binSize < 1) {
+			throw new InvalidMaskingConfigurationException("`binSize` must be greater than 0");
+		}
+		 if (singleBucketOverThresholdReplacement == null) {
+		      throw new InvalidMaskingConfigurationException("`singleBucketOverThresholdReplacement` must not be null");
+		 }
+		 if (singleBucketUnderThresholdReplacement == null) {
+		      throw new InvalidMaskingConfigurationException("`singleBucketUnderThresholdReplacement` must not be null");
+		 }
+		 if (singleBucketOverThresholdValue < 1) {
+				throw new InvalidMaskingConfigurationException("`singleBucketOverThresholdValue` must be greater than 0");
+			}
+		 if (singleBucketUnderThresholdValue < 1) {
+				throw new InvalidMaskingConfigurationException("`singleBucketUnderThresholdValue` must be greater than 0");
+			}
+		if (useSingleBucketOverThreshold == true && useSingleBucketUnderThreshold == true) {
+			if (singleBucketOverThresholdValue < singleBucketUnderThresholdValue) {
+				throw new InvalidMaskingConfigurationException("`singleBucketOverThresholdValue` must be greater than singleBucketUnderThresholdValue");

```

Comment:
The message should say "greater than or equal to", which is what your comparison already does.  I would also wrap the name singleBucketUnderThresholdValue in backticks - `singleBucketUnderThresholdValue`,